pkg/notification/client.go

```diff
@@ -0,0 +1,159 @@
+package notification
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"github.com/opendevstack/pipeline/internal/kubernetes"
+	"github.com/opendevstack/pipeline/pkg/logging"
+	"github.com/opendevstack/pipeline/pkg/pipelinectxt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	NotificationConfigMap   = "ods-notification"
+	UrlProperty             = "url"
+	MethodProperty          = "method"
+	ContentTypeProperty     = "contentType"
+	RequestTemplateProperty = "requestTemplate"
+	NotifyOnStatusProperty  = "notifyOnStatus"
+)
+
+type Client struct {
+	clientConfig     ClientConfig
+	httpClient       *http.Client
+	kubernetesClient kubernetes.ClientInterface
+}
+
+type ClientConfig struct {
+	Namespace string
+	Logger    logging.LeveledLoggerInterface
+}
+
+type PipelineRunResult struct {
+	PipelineRunURL string
+	OverallStatus  string
+	ODSContext     *pipelinectxt.ODSContext
+}
+
+type notificationConfig struct {
+	url            string
+	method         string
+	contentType    string
+	notifyOnStatus []string
+	template       *template.Template
+}
+
+func NewClient(config ClientConfig, kubernetesClient kubernetes.ClientInterface) (*Client, error) {
+	if config.Logger == nil {
+		config.Logger = &logging.LeveledLogger{Level: logging.LevelError}
+	}
+
+	return &Client{
+		clientConfig:     config,
+		httpClient:       &http.Client{},
```

Comment:
I think we should pass this as a dependency. Check the Nexus client, which does sth. similar. In any case, we should set some timeout on the client (because there is no timeout by default!)